amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/bulk/hold/HoldBulkServiceImpl.java

```diff
@@ -0,0 +1,242 @@
+import org.alfresco.module.org_alfresco_module_rm.bulk.BulkBaseService;
+import org.alfresco.module.org_alfresco_module_rm.bulk.BulkOperation;
+import org.alfresco.module.org_alfresco_module_rm.bulk.BulkStatusUpdater;
+import org.alfresco.module.org_alfresco_module_rm.capability.CapabilityService;
+import org.alfresco.module.org_alfresco_module_rm.capability.RMPermissionModel;
+import org.alfresco.module.org_alfresco_module_rm.hold.HoldService;
+import org.alfresco.repo.batch.BatchProcessWorkProvider;
+import org.alfresco.repo.batch.BatchProcessor.BatchProcessWorker;
+import org.alfresco.repo.security.authentication.AuthenticationUtil;
+import org.alfresco.repo.security.permissions.AccessDeniedException;
+import org.alfresco.rest.api.search.model.Query;
+import org.alfresco.rest.framework.core.exceptions.InvalidArgumentException;
+import org.alfresco.rm.rest.api.model.HoldBulkStatus;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.search.ResultSet;
+import org.alfresco.service.cmr.search.SearchParameters;
+import org.alfresco.service.cmr.security.AccessStatus;
+import org.alfresco.service.cmr.security.PermissionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.extensions.surf.util.I18NUtil;
+
+/**
+ * Implementation of the {@link HoldBulkService} interface.
+ */
+public class HoldBulkServiceImpl extends BulkBaseService<HoldBulkStatus> implements HoldBulkService
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(HoldBulkServiceImpl.class);
+
+    private HoldService holdService;
+    private static final String MSG_ERR_ACCESS_DENIED = "permissions.err_access_denied";
+
+    private CapabilityService capabilityService;
+    private PermissionService permissionService;
+    private NodeService nodeService;
+
+    @Override
+    protected HoldBulkStatus getInitBulkStatus(String processId, long totalItems)
+    {
+        return new HoldBulkStatus(processId, null, null, 0, 0, totalItems, null);
+    }
+
+    @Override
+    protected BulkStatusUpdater getBulkStatusUpdater()
+    {
+        return new HoldBulkStatusUpdater((HoldBulkMonitor) bulkMonitor);
+    }
+
+    @Override
+    protected BatchProcessWorkProvider<NodeRef> getWorkProvider(BulkOperation bulkOperation, long totalItems,
+        BulkStatusUpdater bulkStatusUpdater)
+    {
+        return new AddToHoldWorkerProvider(new AtomicInteger(0), bulkOperation, totalItems, bulkStatusUpdater);
+    }
+
+    @Override
+    protected BatchProcessWorker<NodeRef> getWorkerProvider(NodeRef nodeRef, BulkOperation bulkOperation)
+    {
+        if (ADD.name().equals(bulkOperation.operationType()))
+        {
+            return new AddToHoldWorkerBatch(nodeRef);
+        }
+        throw new InvalidArgumentException(
+            "Unsupported action type when starting the bulk process: " + bulkOperation.operationType());
+    }
```

Comment:
Enums are primary targets to use `switch` operators instead of `if-else` blocks.

Reply:
Since we have only one case, IDE suggested to use the `if-else` block for now. I refactored it to use `switch`.